### PR TITLE
Updating SAS Version 

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Updating the SAS Version won't affect any other string-to-string algorithms and is needed for UserDelegationSAS
(Fixes https://github.com/Azure/azure-sdk-for-go/issues/19249)

+ fixing comments to UDC file and test

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/sas/query_params.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ const (
 
 var (
 	// Version is the default version encoded in the SAS token.
-	Version = "2019-12-12"
+	Version = "2020-02-10"
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/sas/query_params.go:26]: This value is also used in the `SignWithSharedKey()` APIs. You're sure this won't break them?

Comment (siminsavani-msft) [reply sdk/storage/azblob/sas/query_params.go:26]: I'm confident! The string-to-sign version 2020-12-06 is the one that will need some changes, but 2020-02-10 will fall under this format (https://learn.microsoft.com/en-us/rest/api/storageservices/create-service-sas#version-2018-11-09-and-later) which is our current string-to-sign behavior.

Comment (jhendrixMSFT) [reply sdk/storage/azblob/sas/query_params.go:26]: Thanks for confirming!

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
Could you also please update `azcore` to `v1.1.4` please and add an entry to the changelog in the "other" section? This resolves an issue with incorrect path joining.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**